src/_lib/modules/tabs/tabs.component.ts

```diff
@@ -1,27 +1,76 @@
-import { Component, ContentChildren, QueryList, Input } from '@angular/core';
+import { Component, ContentChildren, QueryList, Input, ElementRef, ViewChild, OnInit, ChangeDetectorRef, OnDestroy } from '@angular/core';
 import { TabDirective } from './tab.directive';
+import { fromEvent, Subscription, Observable, merge } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+import { TabsService } from './tabs.service';
 
 @Component({
     selector: 'nw-tabs',
     template: `
-      <ul class="nav nav-tabs" [ngClass]="tabSizeClass" role="tablist">
-          <ng-content></ng-content>
-      </ul>
-      <div #border class="nav-tabs-active-bar" [ngStyle]="getStyles()"></div>
+        <div class="scroll-container" #scrollContainer>
+            <ul class="nav nav-tabs" [ngClass]="tabSizeClass" role="tablist">
+                <ng-content></ng-content>
+                <li #border class="nav-tabs-active-bar" [ngStyle]="getActiveStyles()"></li>
+            </ul>
+        </div>
+
+        <div class="pagination-container" *ngIf="shouldShowPagination">
+            <div class="prev-page" *ngIf="shouldShowPrev" (click)="prev()" [ngStyle]="background">
+                <ng-container *ngTemplateOutlet="paginator"></ng-container>
+            </div>
+            <div class="next-page" *ngIf="shouldShowNext" (click)="next()" [ngStyle]="background">
+                <ng-container *ngTemplateOutlet="paginator"></ng-container>
+            </div>
+        </div>
+
+        <ng-template #paginator>
+            <button class="btn btn-md btn-ghost-alt">
+                <i class="fas fa-chevron-left"></i>
+            </button>
+        </ng-template>
   `,
+  providers: [TabsService],
     styles: [`
         :host, ul {
             position: relative;
         }
   `]
 })
 
-export class TabsComponent {
+export class TabsComponent implements OnInit, OnDestroy {
 
     @Input() size: 'sm' | 'md' | 'lg' = 'md';
+    @Input() maskColor: string = '#373737';
+
+    @ViewChild('scrollContainer') scrollContainer: ElementRef<HTMLElement>;
 
     @ContentChildren(TabDirective) tabs: QueryList<TabDirective> = new QueryList();
 
+    private _scrollAndResizeSub: Subscription;
+    private _activeChangeSub: Subscription;
+    private _paginationTolerance: number = 100;
+
+    constructor(
+        private _cdRef: ChangeDetectorRef,
+        private _tabsService: TabsService) {}
+
+    ngOnInit() {
+        const scrollStop$: Observable<Event> = fromEvent(this.scrollContainer.nativeElement, 'scroll');
+        const windowResizeStop$: Observable<Event> = fromEvent(window, 'resize');
```

Comment:
Should these maybe be names scrollEvent$ & windowResizeEvent$ ? Unless I misunderstand their purpose...

Reply:
I chose to name them with the suffix `stop` because of the debounce that is applied. Assuming someone is scrolling and resizing normally these observable will really only emit when the user stops.